### DISCUSSION TITLE
How to clear everything in order to avoid: "different object was already registered with this name, since this is undefined behavior"

### DISCUSSION THREAD

**Discussion by DaveDKAF** [KratosMultiphysics/Kratos#12552] [Q&A]
With regards to the code snippet below I call this in a time loop - I get an intermittent error  

KratosFortran: Setting up Kratos
KratosFortran: Importing MeshingApplication
Initializing KratosMeshingApplication...
terminate called after throwing an instance of 'Kratos::Exception'
  what():  Error: An object of different type was already registered with name "TestElement2D"!

**How do I make sure everything is cleared so this error is avoided?  I use Reset but that isn't enough.**

Code snippet:

```

 Model TMACmodel;
            Kernel mKernel;
            
            // Rest for this time-step
            Kernel::GetApplicationsList().clear();
            TMACmodel.Reset();
            
            KRATOS_INFO("KratosFortran") << "Setting up Kratos" << std::endl;
            
            KratosMeshingApplication::Pointer       mpMeshingApp;
            if (!mKernel.IsImported("MeshingApplication")) {
            KRATOS_INFO("KratosFortran") << "Importing MeshingApplication" << std::endl;
            mpMeshingApp = Kratos::make_shared<KratosMeshingApplication>();
            mKernel.ImportApplication(mpMeshingApp);
            }
            
            ModelPart & r_Model_Part      = TMACmodel.CreateModelPart("Main", 2); // NB: 2 refers to 2D



```

**Comment by DaveDKAF** [comment]
This is a C++ not a python question - thanks!

**Comment by rubenzorrilla** [comment]
Sorry @DaveDKAF. We completely missed this discussion. Is the issue solved?
